### ISSUE TITLE
Grouped convolution produces wrong results on CPU

### ISSUE THREAD

**Issue by andriusstank** [pytorch/pytorch#19912]
## 🐛 Bug
Grouped convolution with 4 input channels, 16 output channels and 2 groups yields wrong result. Interestingly, many other combinations of number of input and output channels work fine.

## To Reproduce
Here's simple 2d 1x1 convolution. One element of input is 1.0, while the rest are zeros. Weight tensor is all ones. However, result on CPU is all zeros.

```python
import torch

inC = 4
outC = 16

x = torch.zeros(1, inC, 1, 1)
x[0,3,0,0] = 1.0
print("x", x)

weight = torch.ones(outC, inC//2, 1, 1)

y1 = torch.nn.functional.conv2d(x, weight, groups=2)
print("y1", y1)
y2 = torch.nn.functional.conv2d(x.cuda(), weight.cuda(), groups=2).cpu()
print("y2", y2)
```
I get the following output:

```
x tensor([[[[0.]],

         [[0.]],

         [[0.]],

         [[1.]]]])
y1 tensor([[[[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]]]])
y2 tensor([[[[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[0.]],

         [[1.]],

         [[1.]],

         [[1.]],

         [[1.]],

         [[1.]],

         [[1.]],

         [[1.]],

         [[1.]]]])
```

## Expected behavior

`y1` and `y2` should be equal.

## Environment

```
PyTorch version: 1.0.1.post2
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Fedora 28 (Twenty Eight)
GCC version: (GCC) 8.3.1 20190223 (Red Hat 8.3.1-2)
CMake version: version 3.14.0

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 418.56
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.14.5
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.3                      199  
[conda] mkl_fft                   1.0.10           py37ha843d7b_0  
[conda] mkl_random                1.0.2            py37hd81dba3_0  
[conda] pytorch                   1.0.1           py3.7_cuda10.0.130_cudnn7.4.2_2    pytorch
[conda] torchvision               0.2.2                      py_3    pytorch
```
## Additional context

There was a bug report about wrong results on CPU with non-contiguous weights that was fixed in Pytorch 1.0.1. All tensors are contiguous here.

**Comment by SsnL** [comment]
Works for me on a master build. Likely fixed somewhere between 1.0.1 and master.

**Comment by soumith** [comment]
it will be part of the 1.1 release tomorrow.
